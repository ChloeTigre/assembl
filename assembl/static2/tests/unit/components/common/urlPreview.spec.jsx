import React from 'react';
import renderer from 'react-test-renderer';
import URLPreview from '../../../../js/app/components/common/urlPreview/urlPreview';

describe('UrlPreview component', () => {
  const imgUrl = 'https://images.unsplash.com/uploads/141155339325423394b24/03982423';

  const authorAvatar =
    'https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260';

  const faviconUrl = 'https://images.pexels.com/photos/247932/pexels-photo-247932.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260';

  const props = {
    id: 'postId',
    url: 'https://foo.bar/1234',
    title: 'My foo&#039;bar title',
    description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been.',
    faviconUrl: faviconUrl,
    authorName: '@tarti',
    authorAvatar: authorAvatar,
    providerName: 'FooBar',
    html: '',
    thumbnailUrl: imgUrl
  };
  it('should match snapshot', () => {
    const rendered = renderer.create(<URLPreview {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});