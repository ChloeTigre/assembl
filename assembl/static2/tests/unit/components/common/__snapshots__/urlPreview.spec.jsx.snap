// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlPreview component should match snapshot 1`] = `
<span
  className="url-container"
>
  <span
    className="url-header"
  >
    <img
      alt="@tarti"
      className="url-contribution-avatar"
      src="https://images.pexels.com/photos/67636/rose-blue-flower-rose-blooms-67636.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260"
    />
    <a
      className="url-contribution-title"
      href="https://foo.bar/1234"
      target="_blank"
    >
      My foo'bar title
    </a>
    <span
      className="url-author-name"
    >
      @tarti
    </span>
  </span>
  <span
    className="url-content-container"
  >
    <span
      className="url-description"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been.
    </span>
    <span
      className="url-media-container"
    >
      <span
        className="url-img-preview"
        style={
          Object {
            "backgroundImage": "url(\\"https://images.unsplash.com/uploads/141155339325423394b24/03982423\\")",
          }
        }
      />
    </span>
  </span>
  <span
    className="url-footer"
  >
    <img
      alt="favicon"
      className="url-footer-avatar"
      src="https://images.pexels.com/photos/247932/pexels-photo-247932.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260"
    />
    <span
      className="url-footer-title"
    >
      FooBar
    </span>
  </span>
</span>
`;
